test/metabase/models/setting_test.clj

```diff
@@ -142,14 +155,54 @@
     (is (= true
            (test-setting-calculated-getter)))
     (is (= true
-           (setting/user-facing-value :test-setting-calculated-getter)))))
+           (setting/user-facing-value :test-setting-calculated-getter))))
+
+  (testing "`user-facing-value` will initialize pending values"
+    (is (some? (setting/user-facing-value custom-init-setting)))))
 
 (deftest do-not-define-setter-function-for-setter-none-test
   (testing "Settings with `:setter` `:none` should not have a setter function defined"
     (testing "Sanity check: getter should be defined"
       (is (some? (resolve `test-setting-calculated-getter))))
     (is (not (resolve `test-setting-calculated-getter!)))))
 
+(defn- get-setting-via-db []
+  (#'setting/read-setting :test-setting-custom-init))
```

Comment:
I'm struggling to find the benefit of defining this function. Again this is subjective but I've noticed you tend to create extra little functions and I prefer to inline them for readability when DRYing doesn't bring any benefits. Up to you again but I wouldn't be so scared of duplication here.

Reply:
In this particular case I think the helper method was created initially to encapsulate fiddling with a dynamic var, and is now a bit vestigial that the `read-setting` method was added. I'll just inline that.

On your general observation - I'm a believer in the the top-down / declarative / single-level-of-abstraction / small methods school of "clean" code. I take it that you subscribe to a more procedural / details clearly exposed style. The both have their merits, and based on familiarity one is either more comfortable drilling between methods or parsing dense code chunks. I agree that "coincidental repetition" should not be duplicated, but otherwise I err on the side of small composable atomic things. For me part of the appeal of FP is that large sections of the code can be self describing aliases essentially, that should never need to change, and might simply get deleted one day.

I'll assimilate the local style as I get more familiar with it - consistency often beats personal taste 😄